extensions/smallrye-health/runtime/src/main/java/io/quarkus/smallrye/health/runtime/SmallRyeHealthHandlerBase.java

```diff
@@ -22,8 +31,14 @@
 
 public abstract class SmallRyeHealthHandlerBase implements Handler<RoutingContext> {
 
+    static boolean problemDetails = false;
```

Comment:
Now that I think of it, this should be `volatile`

Reply:
why please?

Reply:
The reason is that `SmallRyeHealthHandlerBase.problemDetails` is set by the main thread (although there is no strict contract for this, this is how it works in practice), but the field is read on an event loop thread (but potentially by any thread).

Reply:
Just wondering how much of a potentially unnecessary performance cost it has given that it's only written once. This could be avoided by instead passing it to a constructor of the handler. But maybe the performance cost of it being `volatile` is negligible.

Reply:
Obviously the best solution (for more than just performance) is to have final fields.

Given that this is not on the hot path, I wouldn't worry much about the (tiny) performance hit